src/frequenz/dispatch/actor.py

```diff
@@ -114,18 +89,28 @@ async def _fetch(self) -> None:
 
         try:
             _logger.info("Fetching dispatches for microgrid %s", self._microgrid_id)
-            async for dispatch in self._client.list(microgrid_id=self._microgrid_id):
-                self._dispatches[dispatch.id] = dispatch
+            async for client_dispatch in self._client.list(
+                microgrid_id=self._microgrid_id
+            ):
+                dispatch = Dispatch(client_dispatch)
 
+                self._dispatches[dispatch.id] = Dispatch(client_dispatch)
                 old_dispatch = old_dispatches.pop(dispatch.id, None)
                 if not old_dispatch:
                     self._update_dispatch_schedule(dispatch, None)
                     _logger.info("New dispatch: %s", dispatch)
-                    await self._updated_dispatch_sender.send(Created(dispatch=dispatch))
+                    await self._lifecycle_updates_sender.send(
+                        Created(dispatch=dispatch)
+                    )
```

Comment:
What if it is just created but already ready to run? Shouldn't it be sent to the running state change channel too, or will that be handled by a separate task taking care of the scheduling?

Reply:
Yes, this happens in `_schedule_task()` called from two lines higher `_update_dispatch_schedule()`